server/src/main/java/org/opensearch/common/time/DateFormatters.java

```diff
@@ -741,7 +743,9 @@ public class DateFormatters {
             .toFormatter(Locale.ROOT)
             .withResolverStyle(ResolverStyle.STRICT),
         new DateTimeFormatterBuilder().append(STRICT_DATE_FORMATTER)
+            .optionalStart()
```

Comment:
One of the reasons it is called `STRICT_DATE_TIME` is because all components are required, changing that would lead to surprising results. We have a large number of relaxed formats like  STRICT_DATE_OPTIONAL_TIME, STRICT_DATE_OPTIONAL_TIME_NANOS, those are explicit on optionality of some components.

Reply:
Agreed that we probably don't want to change the definition of what `STRICT_DATE_TIME` means.

Reply:
Seems reasonable. In this case I'll close this PR and recommend that the person who created the original issue changes their mapping to use a different format.